playbooks/roles/caasp4/tasks/setup_nodes.yml

```diff
@@ -15,6 +15,9 @@
 # under the License.
 #
 
+- name: Assert required packages are alredy present
+  include: _assert_packages.yml
```

Comment:
pretty sure @toabctl wont agree with this, as it makes the role more dependent on our own stuff instead of being agnostic. The proper place to set up this may be in the playbooks instead of the role (so {openstack,kvm}-deploy_caasp.yml playbooks)

Reply:
What do you mean? We have lot of our own stuff throughout various roles (related to caasp, deployer configuration). 
@Itxaka Can you clarify as I don't follow this idea.

Reply:
I agree with @itxaka here.

I would prefer to see this outside of caasp4.

In fact, I would prefer that we create a role for this, like bootstrap_deployer.

I think it's long due that we clean up code, and this could be one way to do it.

Reply:
the idea is that  roles are kept as agnostic as possible and the m,odifications/prerequisites to those are don on the playbooks calling those roles. For example, we load our vars in the playbook before calling the role, instead of having the role load our custom vars itself. That way we dont tie a role that does X with *only* one way of using it and we can reuse it further for different tasks/change it enough with different vars/prerequisites.

See for example the gather logs roles from upstream openstack. They are not tied to zuul or opendev, we can use them by passing a couple of vars before calling the role. Thats what we want, something that can be reused.

installing those specific packages for a specific OS makes this role very narrow and impossible to reuse.


I hope that was clear?

Reply:
Thanks. Understand the idea now.

Looks like I will need to new role for this as assert_packages has number of tasks and adding them in {openstack,kvm}-deploy_caasp.yml will clutter the top playbook.